iroh-rpc-types/src/gateway.rs

```diff
@@ -34,3 +50,11 @@ impl Service for GatewayService {
 impl RpcMsg<GatewayService> for VersionRequest {
     type Response = VersionResponse;
 }
+
+impl Msg<GatewayService> for WatchRequest {
```

Comment:
Even if we move the WatchRequest / VersionRequest one level up, these would have to be declared separately for each service.